content/guides/http/go/net_http/client.md

```diff
@@ -0,0 +1,279 @@
+
+	// It is imperative that req.WithContext is used to
+	// propagate context and use it in the request.
+	req = req.WithContext(ctx)
+
+	client := &http.Client{Transport: &ochttp.Transport{}}
+	res, err := client.Do(req)
+	if err != nil {
+		log.Fatalf("Failed to make the request: %v", err)
+	}
+
+	// Consume the body and close it.
+	io.Copy(ioutil.Discard, res.Body)
+	_ = res.Body.Close()
+}
+```
+
+and then using a [Trace exporter](/exporters/supported-exporters/go) to examine the traces.
+
+### Metrics
+
+Metrics can be enabled by simply registering the [ochttp.DefaultClientViews](https://godoc.org/go.opencensus.io/plugin/ochttp#DefaultClientViews)
+
+```go
+        // In our main, register ochttp Client views
+        if err := view.Register(ochttp.DefaultClientViews...); err != nil {
+                log.Fatalf("Failed to register client views for HTTP metrics: %v", err)
+        }
+```
+
+and then using a [Stats exporter](/exporters/supported-exporters/go) to examine the metrics.
+
+which provide the following metrics
+
+Metric|Prefix|Description|Tags|Aggregation
+---|---|---|---|---
+Requests completed|"opencensus.io/http/client/completed_count"|The number of requests made|"http_client_method", "http_client_status"|Count
+Bytes sent|"opencensus.io/http/client/sent_bytes"|The number of bytes sent|"http_client_method", "http_client_status"|Distribution
+Bytes received|"opencensus.io/http/client/received_bytes"|The number of bytes received|"http_client_method", "http_client_status"|Distribution
+Roundtrip Latency|"opencensus.io/http/client/roundtrip_latency"|The end-to-end latency|"http_client_method", "http_client_status"|Distribution
+
+### End to end example
+
+The following example uses the ochttp.Transport to make requests to the OpenCensus website every 5 seconds and exports traces and metrics to
+Trace and Metrics exporters: Zipkin and Prometheus respectively. Of course you can use [any other exporters of your choice](/exporters/supported-exporters/go)
+
+{{% notice tip %}}
+For assistance setting up any of the exporters, please refer to:
+
+Exporter|URL
+---|---
+Prometheus|[Prometheus codelab](/codelabs/prometheus)
+Zipkin|[Zipkin codelab](/codelabs/zipkin)
+{{% /notice %}}
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	openzipkin "github.com/openzipkin/zipkin-go"
+	zipkinHTTP "github.com/openzipkin/zipkin-go/reporter/http"
+
+	"go.opencensus.io/exporter/prometheus"
+	"go.opencensus.io/exporter/zipkin"
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+)
+
+func main() {
+	// Firstly, we'll register ochttp Client views
+	if err := view.Register(ochttp.DefaultClientViews...); err != nil {
+		log.Fatalf("Failed to register client views for HTTP metrics: %v", err)
+	}
+
+	// Enable observability to extract and examine traces and metrics.
+	enableObservabilityAndExporters()
```

Comment:
Since we have view registration be inside main it would be cleaner to take out trace.ApplyConfig and place it at main function level too... this would mean this function would be taking on only once concern which is enabling exporters and not generic OpenCensus instrumentation configuration.

Reply:
Great idea, I also thought about that, thanks for confirming the idea.